tests/Doctrine/Tests/ORM/Query/SelectSqlGenerationTest.php

```diff
@@ -2242,6 +2259,43 @@ public function testHavingSupportResultVariableInAggregateFunction()
             'SELECT COUNT(c0_.name) AS sclr_0 FROM cms_users c0_ HAVING sclr_0 IS NULL'
         );
     }
+
+    /**
+     * @group DDC-3701
+     */
+    public function testFROMClauseAcceptsFQCNWithLeadingBackslash()
+    {
+        $this->_em->createQuery(
+            'SELECT c FROM \Doctrine\Tests\Models\Company\CompanyContract c'
```

Comment:
What I was trying to say was that it *seems* that for the rest of the code the leading `\` makes no difference. And that there are some tests actually using it, but maybe the authors would have noticed and changed that *if* the parser had actually complained.